Design Explorer 2

MingboPeng opened the pull request:
Added google Link, and full screen charts.
New scatter plots chart.

MingboPeng committed:
connect Google

load data from google sheet and folder

MingboPeng committed:
overwrite the rest files that may have changes

overwrite the rest files that may have changes from DE1

MingboPeng committed:
Fix google link input css

Fix google link input css

MingboPeng committed:
Fix bugs of uploading csv after used google sheets

user can upload csv file after used google sheet without reload the
page.

MingboPeng committed:
added acatter matrix source

added acatter matrix source from benjiec

MingboPeng committed:
Added new Scatter Chart on right side

New scatter charts, and set charts full screen.

MingboPeng committed:
parallel coordinates full size available

added button to set parallel chart full size, but there is a bug of
missing brush after resize. I need to fix it next step.

MingboPeng committed:
left chart full screen tested with new parcoords.js

It seems has a bug that cannot resize the height of left chart, need to
compare with the previous parcoords.js version

MingboPeng committed:
fixed the most major functional bugs when using new parcoords js.

All charts full screen basically working now, but still there are a lot
of minor bugs left.

MingboPeng committed:
Merge pull request #1 from MingboPeng/left_chart_fullscreen

Left chart fullscreen

MingboPeng committed:
fixed: zoomed area coordinates with the parallel chart when it is switching full screen mode

1. fixed: zoomed area coordinates with the parallel chart when it is
switching full screen mode
2. fixed: clean the pre-parameters before load new data
3. beautify the general code structures.
4. correct codes' miner typos

MingboPeng committed:
fixed some workflow presenting issues

MingboPeng committed:
Improve the scatter chart presentation

the dots size, opacity

MingboPeng committed:
fixed the rating scale problem

in parcoords.js the function scale seems run before autoscale, so that
the value will be overwrite by autoscale.. might need to double check
later.

MingboPeng committed:
fixed thumbnails-btm_container's height

fixed thumbnails-btm_container's height when
isLeftChartFullScreenToggled

mostaphaRoudsari commented:
@MingboPeng thank you for the pull request. I know that I asked you to send a pull request to this repository but at the same time I'm not sure who is going to make decisions about the future of DesignExplore at this point. Let me email @bhowes-tt and @JonatanS to see what will be the best approach and I will get back to you.

mostaphaRoudsari commented:
@MingboPeng, @bhowes-tt will take care of this PR. Also make sure that you guys will have lunch when you start working at TT soon. Thanks again for doing this :100:

bhowes-tt commented:
@MingboPeng - thank you so much for this.  Awesome work!  :boat: 

I'm reviewing / testing now, and will likely come back with a few small items for you before we merge this pull.  I'm excited to meet you!  Be sure to come find us when you get started at TT.

More to follow - stay tuned.

bhowes-tt commented:
@MingboPeng 

I was able to get my own example running using your new build, and google drive to host and serve the images and the data ... AMAZING WORK  :boat: :octopus: :lemon: !!!

 There are a few issues that need your attention before we'll merge the pull, however:

1)  The major issue is the 3D viewer / links to spectacles.json files.  I added a 'threeD' column to my spreadsheet, and uploaded .json files exported from Grasshopper using Spectacles.  I figured the 3D content would/should work just like the image content, but it looks like your current build is looking for the .json files on the web server, not in the google drive folder.  Also, it looks like you are getting a vA3C / Spectacles error:
![image](https://cloud.githubusercontent.com/assets/3171551/16532898/fb0a457c-3fa2-11e6-8306-59c88ce39f84.png)

Please re-enable the 3D viewer, and let Spectacles.json files work just like images do in the google folder/spreadsheet.

2)  Sliders are still there and working when an old sample data set is loaded (like the one that loads by default), but don't show up in my example.  Is this intentional or a bug?  Please explain... Maybe I'm missing something...
![image](https://cloud.githubusercontent.com/assets/3171551/16533122/916512b2-3fa4-11e6-9a60-1d9a4bca7df3.png)
![image](https://cloud.githubusercontent.com/assets/3171551/16533154/bf9eaa80-3fa4-11e6-8040-ecff4b1284e4.png)

3) If the parallel coordinates UI is hidden with the little down arrow, it can never be brought back.  This is working in our old build.  Please fix this if you can (this is by far the lowest priority issue on this list).
![image](https://cloud.githubusercontent.com/assets/3171551/16533227/369f7470-3fa5-11e6-8f33-4f2f15afb1f4.png)

4) Logo / Splash page.  Let's keep the name of the project the same - 'Design Explorer', not 'Design Explorer 2' - and revert back to the old logo.  I'd also ask that we omit some of the content on the splash view - see the image below.  You can certainly keep all of that content on your fork if you'd like to, but I don't think it makes sense as part of a pull request.
![image](https://cloud.githubusercontent.com/assets/3171551/16533578/c4f2e3e0-3fa7-11e6-86c9-81ea7e2d4061.png)

5)  Scatterplots.  I do prefer the scatterplots over the radar chart...  But, if I check anything in the horizontal list, the plots disappear - it looks like it's just using [2 or more] vertical variables, which feels like a bug.  Please have a look and either let us specify both horizontal and vertical (which we'd prefer), or just provide a single list.  Also, please refer to the library you are using in the .README file.

Please let me know if you have any questions about any of these issues, and thank you so much for the great work.  I'm super excited about the pull... very well done.

MingboPeng commented:
Hi Ben,

It seems cannot reply this by email!

Thanks for testing out.

1) The google drive folder was built for images originally, but I have found the method to load json from google drive file as well. Working on it.
2) The slider problem is because you didn’t put “in:” or “out:” in the csv or google. I also have a solution for it. 
3) This is strange, I will fix it.
4) Sure, the logo and other staffs were used in the one hosted in a MEBD related server (for UPenn student use purpose), and definitely happy to remove some unnecessary content. 
5) This is Horizontal axis only works while checked item is also checked in the Vertical axis. I know this is weird, because of the way used in original library. I think the easiest solution might be remove all horizontal items, and change vertical axis to the single list. 

Thanks again.
Mingbo

MingboPeng committed:
Load Json from Google Drive and fixed bugs

Fixed bugs:
1. Hide/Show Parallel coordinate
2. slider's img and 3d link problem

MingboPeng committed:
Added Scatter-matrix Chart reference

Added Scatter-matrix Chart reference

bhowes-tt commented:
Excellent.  I'll give the new build another test today...

bhowes-tt commented:
@MingboPeng - Looking good!  I like the new loading sequence.

no luck today though, maybe you are still in process, or I'm doing something wrong.  I'm getting a http request error after pasting my folder link:
![image](https://cloud.githubusercontent.com/assets/3171551/16743660/6c8b7f24-477b-11e6-9151-79d950fd2978.png)

I tried uploading data.csv, and creating and publishing a sheet called data.csv - both produced the same error.

MingboPeng commented:
@bhowes-tt because google drive now needs different settings than before, I will make a tutorial soon.

MingboPeng committed:
Supported Google Drive and server link, and clean the code

Supported Google Drive and server link, and clean the code

MingboPeng commented:
@bhowes-tt  here is a [tutorial demo](http://www.mpendesign.com/tutorial/tutorial-and-demo-for-design-explorer-2-2/)

bhowes-tt commented:
@MingboPeng - sorry for the long radio silence.  I finally got around to testing your latest version, and we are good to go!  :boat: :octopus: :boat: 

Thank you again for the really great work.  We're going to publish a blog post about this project soon, and I'd like to get some quotes from both you and @mostaphaRoudsari .

MingboPeng commented:
@bhowes-tt Thanks so much for your testing. Now I am working on a new tutorial of how to prepare the data from Grasshopper for Design Explorer. Trying to make it simple for everyone to use.
Thanks again,
Mingbo
